e2e/testcases/apiservice_test.go

```diff
@@ -57,9 +63,6 @@ func TestCreateAPIServiceAndEndpointInTheSameCommit(t *testing.T) {
 	nt.WaitForRepoSyncs()
 
 	nt.RootRepos[configsync.RootSyncName].Remove("acme/namespaces/custom-metrics/namespace-custom-metrics.yaml")
```

Comment:
If you don't plan to commit the change, you should remove `nt.RootRepos.Remove` as well.

Reply:
It is getting committed in the second line from here.

Reply:
Discussed: let Config Sync remove the test APISerivce first so it's not blocked by Discovery failure later when cleanup rest of repo.